internal/trafficcontrol/tc.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trafficcontrol
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/vishvananda/netlink"
+)
+
+var (
+	sysClassNet = "/sys/class/net/"
+	tcpProtoStr = "tcp"
+	udpProtoStr = "udp"
+)
+
+func AllowTrafficFromMac(ifName string, macAddress string) error {
+	if _, err := exec.LookPath("tc"); err != nil {
+		return err
+	}
+	interfaces, err := GetAssociatedInterfaces(ifName)
+	if err != nil {
+		return err
+	}
+	for _, iface := range interfaces {
+		cmd := exec.Command("bash", "-c", fmt.Sprintf("tc filter replace dev %s ingress pref 9000 protocol all flower src_mac %s action ok", iface, macAddress))
```

Comment:
I think we need to eliminate bash from all of these exec commands.  If you use `/usr/sbin/tc` instead of `tc`, does that work?

That said, I'm okay with that kind of cleanup being in a separate PR; no need to block this one on that change provided we plan to clean this up in the future.